### PR TITLE
feat(invoices): show paid and balance columns into the invoices list

### PR DESCRIPTION
**Pull request by jhoffman-ms**
Added Paid and Balance columns  into the invoices list:

image

**Changes:**

- Added Paid and Balance columns  into the invoices list

- Added new unit tests

**Task:** [DAT-157](https://makingsense.atlassian.net/jira/software/projects/DAT/boards/62?selectedIssue=DAT-157)

### CODE REVIEW

File: src/services/doppler-billing-api-client.ts [hunk 1]
```diff
@@ -83,6 +96,8 @@ export class HttpDopplerBillingApiClient implements DopplerBillingApiClient {
       dueDate: !!x.dueDate ? new Date(x.dueDate) : null,
       currency: x.currency,
       amount: x.amount,
+      paidToDate: !!x.paidToDate ? x.paidToDate : 0,
+      balance: this.getBalance(x),
```
Comment (andresmoschini) [inline src/services/doppler-billing-api-client.ts:100]: Is it really necessary? I mean, if you do not have the `balance` field you neither have the `paidToDate` field.

So, in my opinion, it should be simply: 

```suggestion
      balance: data.balance == undefined ? x.amount : x.balance
```

or 

```suggestion
      balance: data.balance == undefined ? null : x.balance
```

Comment (andresmoschini) [reply src/services/doppler-billing-api-client.ts:100]: I mean, I think that the backend is responsible for the balance value.

Comment (jhoffman-ms) [reply src/services/doppler-billing-api-client.ts:100]: Perfect, I like it:

```balance: data.balance == undefined ? x.amount : x.balance```

I will replace for it.

Comment (andresmoschini) [reply src/services/doppler-billing-api-client.ts:100]: ok, please test it, I did not.

Comment (jhoffman-ms) [reply src/services/doppler-billing-api-client.ts:100]: yes, totally agree

Comment (jhoffman-ms) [reply src/services/doppler-billing-api-client.ts:100]: @andresmoschini, I will use: ```balance: !!x.balance ? x.balance :  x.amount,``` to be consistent with the other statements. Do you agree?

Comment (jhoffman-ms) [reply src/services/doppler-billing-api-client.ts]: Done

**Comment by cbernat** [comment]
image
What about this?

**Comment by jhoffman-ms** [comment]
Thanks @cbernat , I will review it.

**Comment by jhoffman-ms** [comment]
@cbernat, the problem is because the header for it "download" column is empty. Maybe we need set a header for example : "Actions". @GusBaamonde/ @cbernat  do you have any suggestions?.

**Comment by GusBaamonde** [comment]
We should not have empty headers, if the action is download, that should be our **"Downloads"** header. If later we have more shares we can change the name, what do you think?

**Comment by jhoffman-ms** [comment]
Thanks @GusBaamonde. For me is Ok. I will add it in the header.

**Comment by jhoffman-ms** [comment]
Done.

image

**Comment by jhoffman-ms** [comment]
Do you have another feedback or suggestion or is ready to merge?

**Pull request merged**